activation/nipost.go

```diff
@@ -291,6 +289,21 @@ func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.NIPos
 		}
 
 		nb.persistState()
+
+		post := nipost.Post{
+			Nonce:         proof.Nonce,
+			Indices:       proof.Indices,
+			Pow:           proof.Pow,
+			Challenge:     nb.state.PoetProofRef[:],
+			NumUnits:      postInfo.NumUnits,
+			CommitmentATX: postInfo.CommitmentATX,
+		}
+		if postInfo.Nonce != nil {
+			post.VRFNonce = *postInfo.Nonce
+		}
+		if err := nipost.AddPost(nb.localDB, nb.signer.NodeID(), post); err != nil {
+			nb.log.Warn("failed to persist post", zap.Error(err))
+		}
```

Comment:
We should align on this, I don't think every post needs to be persisted on every `BuildNiPost`.

Reply:
We need to always have _some_ POST persisted to be able to recertify after a checkpoint recovery because the checkpoint recovery removes the NiPoST field from all existing ATXs.

Reply:
See my other comments regarding `initial_post` and migrations. I just don't think we need to ever update the post that we persist for certificate purposes.